lib/unexpectedMitm.js

```diff
@@ -875,327 +597,26 @@ module.exports = {
-                             * mock is seen.
-                             *
-                             * Instead simply record any exchanges that
-                             * passed through and defer the resolution or
-                             * rejection to the handler functions attached
-                             * to the delegated assertion execution below.
-                             */
-                        }
-
-                        function deliverMockResponse(mockResponse, mockResponseError) {
-                            setImmediate(function () {
-                                var nonEmptyMockResponse = false;
-                                if (mockResponse) {
-                                    res.statusCode = mockResponse.statusCode;
-                                    mockResponse.headers.getNames().forEach(function (headerName) {
-                                        mockResponse.headers.getAll(headerName).forEach(function (value) {
-                                            nonEmptyMockResponse = true;
-                                            res.setHeader(headerName, value);
-                                        });
-                                    });
-                                    var unchunkedBody = mockResponse.unchunkedBody;
-                                    if (typeof unchunkedBody !== 'undefined' && unchunkedBody.length > 0) {
-                                        nonEmptyMockResponse = true;
-                                        res.write(unchunkedBody);
-                                    } else if (nonEmptyMockResponse) {
-                                        res.writeHead(res.statusCode || 200);
-                                    }
-                                }
-                                if (mockResponseError) {
-                                    setImmediate(function () {
-                                        clientSocket.emit('error', mockResponseError);
-                                        assertMockResponse(mockResponse, mockResponseError);
-                                    });
-                                } else {
-                                    res.end();
-                                }
-                            });
-                        }
-
-                        /*
-                         * hook final write callback to record raw data and immediately
-                         * assert request so it occurs prior to request completion
-                         */
-                        observeResponse(res).then(function (rawBuffer) {
-                            var mockResponse = createMockResponse(rawBuffer);
-                            assertMockResponse(mockResponse);
-                        });
-                    }));
+                var mocker = new UnexpectedMitmMocker({
+                    descriptions: requestDescriptions
```

Comment:
Could this either be `descriptions` or `requestDescriptions` in both places? I think I'd prefer `requestDescriptions`. Or how about `expectedExchanges`?

Reply:
Agreed the inconsistency doesn't make much sense.

As for the specific name, I think part of the difficult is it's the expected request but an arranged response. I think I'll go with `requestDescriptions` for now and if we iterate we can do another pass and change it throughout.